src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/AppleCertificatePal.cs

```diff
@@ -399,7 +399,21 @@ private void EnsureCertData()
                 return;
 
             Debug.Assert(!_certHandle.IsInvalid);
-            _certData = new CertificateData(Interop.AppleCrypto.X509GetRawData(_certHandle));
+            string? subjectSummary = Interop.AppleCrypto.X509GetSubjectSummary(_certHandle);
+
+            try
+            {
+                _certData = new CertificateData(Interop.AppleCrypto.X509GetRawData(_certHandle));
+            }
+            catch (CryptographicException e) when (subjectSummary != null)
+            {
+                string message = SR.Format(
```

Comment:
I feel like we started using exception filters, and then stopped again for some reason (AoT, maybe?)

Since I can't remember, and can't think of who to ask pre-coffee, let's just go ahead and do it the classic way.

```suggestion
            catch (CryptographicException e)
            {
                if (subjectSummary is null)
                {
                    throw;
                }
                
                string message = SR.Format(
```